File: Data Structures & Algorithms/jump-game-ii/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    jump(nums) {
        let n = nums.length ;
        const dfs= i=>{
            if(i>=n) return 100000;
            if(i==n-1) return 0;
            let res = 100000;
            for(let j=1;j<=nums[i];j++){
                res=Math.min(res,dfs(i+j)+1);
            }
            return res ;
        }
        return dfs(0);
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-4.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i=0
        n=len(nums)
        if n==1 :
            return n
        res=[]
        while i<n-1:
            while i<n-1 and nums[i]==nums[i+1]:
                i+=1
            res.append(nums[i])
            i+=1
        if i==n-1 and nums[i-1]!=nums[i]:
            res.append(nums[i])
        for i in range(len(res)):
            nums[i]=res[i]
        return len(res)